tensorflow/core/common_runtime/dml/dml_command_recorder.h

```diff
@@ -55,17 +56,17 @@ class DmlCommandRecorder {
                           _Outptr_ ID3D12Fence** fence,
                           _Out_ uint64_t* completion_value);
 
-  Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> GetCommandList();
-
   void ResourceBarrier(absl::Span<const D3D12_RESOURCE_BARRIER> barriers);
-  void AddUAVBarrier();
 
   void CloseAndExecute();
 
   // If false, there are no pending commands to be submitted which indicates
   // that CloseAndExecute() would be a no-op.
   bool HasUnflushedWork() const;
 
+  Status GetStatus() const { return status_; }
```

Comment:
Hmm, it seems like it'd be simpler to just return the Status from CloseAndExecute and ExecuteCommandList. That way you wouldn't need these Get/Reset methods.

Reply:
We still need a way to reset the status when flushing to let the user try something else after catching the exception. Otherwise, the status will permanently be in a failed state. We can't reset it in CloseAndExecute() since it may be called multiple times before we get a chance to flush.